### ISSUE TITLE
[CAL-3893] Error after build: iframe doesn't exist. `createIframe` must be called before `doInIframe`

### ISSUE THREAD

**Issue by fabriziospadaro** [calcom/cal.com#15357]
### Issue Summary

When embedding an iframe in a Next.js page, an error occurs after building the page: `iframe doesn't exist. createIframe must be called before doInIframe`. This issue does not occur during development (`npm run dev`).
Note that the iframe embed still works, and all the functionalities are preserved.
### Steps to Reproduce

1. Create a Next.js page and embed an iframe using `@calcom/embed-react`.
2. Use the following code snippet in your component:
    ```jsx
    "use client"

    import Cal, { getCalApi } from "@calcom/embed-react";
    import { useEffect } from "react";

    export default function CalWidget({ calLink }) {
      useEffect(() => {
        (async function () {
          const cal = await getCalApi();
          cal("ui", {
            "theme": "light",
            "styles": { "branding": { "brandColor": "#2B5ADC" } },
            "hideEventTypeDetails": false,
            "layout": "month_view",
          });
        })();
      }, [])

      return (
        <Cal
          id="book"
          onScrollCapture={(e) => {
            e.stopPropagation();
          }}
          calLink={calLink}
        />
      )
    }
    ```
3. Run the development server using `npm run dev` and verify that the page works as expected.
4. Build the project using `npm run build` and then start the server using `npm start`.
5. Navigate to the page and observe the error message.

### Actual Results

- After building the page and starting the server, the error message `iframe doesn't exist. createIframe must be called before doInIframe` is displayed.
- The iframe fails to load correctly.

### Expected Results

- The iframe should load without any errors, similar to the behavior during development (`npm run dev`).
- The embedded content should be displayed as expected.

<sub>[CAL-3893](https://linear.app/calcom/issue/CAL-3893/error-after-build-iframe-doesnt-exist-createiframe-must-be-called)</sub>

**Comment by alisayed95** [comment]
I'm facing the same issue and I'm not able to solve. any help here?

**Comment by alisayed95** [comment]
@dosu the proposed solution is not working

**Comment by wallawe** [comment]
@dosu you left code that is irrelevant to the example provided. the snippet we are given as developers is as the OP mentioned:
```
  useEffect(() => {
    (async function () {
      const cal = await getCalApi();
      cal("ui", {
        "theme": "light",
        "styles": { "branding": { "brandColor": "#2B5ADC" } },
        "hideEventTypeDetails": false,
        "layout": "month_view",
      });
    })();
  }, [])
  ```
  
  there are no `createIframe` or `doInIframe` methods being used or provided in the snippet that cal.com gives us.

**Comment by fabriziospadaro** [comment]
Any update for this? My sentry is going crazy 😮‍💨

**Comment by fabriziospadaro** [comment]
@dosu The correct api to use should be:

` cal("on", {
        action: "__iframeReady",
        callback: (e) => {
          cal("ui",
            {
              "theme": "light",
              "styles": {
                "branding": { "brandColor": "#2B5ADC" },
              },
              "layout": "month_view",
            });
        }
      });`
      
but this still doesn't fix the issue.

**Comment by hershaw** [comment]
I was getting this error both in dev and the built server but with all functionality working. Looking at the iframe after rendering, I see the the following:

```html
<iframe class="cal-embed" name="cal-embed=" .../>
```

with the name attribute looking like there's something missing on the `name`.

Not even sure this is the relevant code but it looks like [it should be set to namespace](https://github.com/calcom/cal.com/blob/main/packages/embeds/embed-core/src/embed.ts#L277).

So I added the namespace parameter to my code and the error went away:

```ts
useEffect(() => {
    (async function () {
      const cal = await getCalApi({ namespace: "nynamespace" });
      cal("ui", {
        theme: "light",
        styles: {
          branding: { brandColor: "white" },
        },
      });
    })();
  }, []);
```

Buuuuuuut the iframe still has `name="cal-embed="`.

@dosu what do you think?

**Comment by fabriziospadaro** [comment]
Correct, adding the namespace in the initialization, and adding it also to the jsx element fixed the error!
